tests/runtime/work_management/test_abortable_context.py

```diff
@@ -0,0 +1,162 @@
+# Copyright The Caikit Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# Standard
+from concurrent.futures import Future, ThreadPoolExecutor
+import dataclasses
+import datetime
+import random
+import time
+
+# Third Party
+import grpc
+import pytest
+
+# Local
+from caikit.runtime.types.aborted_exception import AbortedException
+from caikit.runtime.work_management.abortable_context import (
+    AbortableContext,
+    ThreadInterrupter,
+)
+from caikit.runtime.work_management.rpc_aborter import RpcAborter
+from tests.fixtures import Fixtures
+
+
+@pytest.fixture(scope="session")
+def thread_interrupter():
+    interrupter = ThreadInterrupter()
+    interrupter.start()
+
+    yield interrupter
+
+    interrupter.stop()
+
+
+@pytest.fixture()
+def grpc_context() -> grpc.ServicerContext:
+    return Fixtures.build_context("abortable-context-test")
+
+
+@pytest.fixture()
+def rpc_aborter(grpc_context):
+    return RpcAborter(grpc_context)
+
+
+def wait_for_interrupter_to_run(interrupter, timeout=1):
+    """Helper to wait until the interrupter's queue is empty.
+    This should only deadlock if the interrupter's polling thread exits.
+    """
+    start = datetime.datetime.now()
+    while (datetime.datetime.now() - start).total_seconds() < timeout:
+        if interrupter._queue.empty():
+            return
+        time.sleep(0.001)
+
```

Comment:
NIT: In other tests, I tend to like putting a comment bar separating `## Helpers #####...` from `## Tests #####...`. Personal preference, but I think it makes the test files more readable and organized.